src/Device/deviceSetup.js

```diff
@@ -196,12 +207,35 @@ export const setupDevice =
                 device,
                 deviceSetupConfig
             );
-            doStartWatchingDevices();
-            dispatch(deviceSetupComplete(preparedDevice));
-            onDeviceIsReady(preparedDevice);
+            onSuccessfulDeviceSetup(
+                dispatch,
+                preparedDevice,
+                doStartWatchingDevices,
+                onDeviceIsReady
+            );
         } catch (error) {
             dispatch(deviceSetupError(device, error));
-            if (!deviceSetupConfig.allowCustomDevice) {
+            if (
+                deviceSetupConfig.allowCustomDevice &&
+                error.message.includes('No firmware defined')
+            ) {
+                logger.info(
+                    `Connected to device with serial number: ${device.serialNumber} ` +
+                        `and family: ${device.deviceInfo.family || 'Unknown'} `
+                );
+                logger.debug(
+                    'Note: no pre-compiled firmware is available for the selected device. ' +
+                        'You may still use the app if you have programmed the device ' +
+                        'with a compatible connectivity firmware.'
```

Comment:
I'd change it from `compatible connectivity firmware` to just `compatible firmware` since this message can be shown in multiple apps (e.g. DTM)

Reply:
Good idea. I mindlessly [copied it from the BLE app](https://github.com/NordicSemiconductor/pc-nrfconnect-ble/blob/57388a4451a3c21a2b48462b07263d37e2968520/index.jsx#L129-L133) but your wording is better. Will change that.

Reply:
I will also change the log level from `debug` to `info`, because I consider this message to be helpful for end users, especially struggling to understand why an app that supports custom devices does not work with their device.